doc/examples_sphinx-gallery/stochastic_variability.py

```diff
@@ -0,0 +1,93 @@
+"""
+.. _tutorials-stochastic-variability:
+
+=========================================================
+Stochastic Variability in Community Detection Algorithms
+=========================================================
+
+This example demonstrates the variability of stochastic community detection methods by analyzing the consistency of multiple partitions using similarity measures (NMI, VI, RI) on both random and structured graphs.
+
+"""
+# %%
+# Import Libraries
+import igraph as ig
+import numpy as np
+import matplotlib.pyplot as plt
+import itertools
+
+# %%
+# First, we generate a graph.
+# Generates a random Erdos-Renyi graph (no clear community structure)
+def generate_random_graph(n, p):
+    return ig.Graph.Erdos_Renyi(n=n, p=p)
+  
+# %%
+# Generates a clustered graph with clear communities using the Stochastic Block Model (SBM)
+def generate_clustered_graph(n, clusters, intra_p, inter_p):
+    block_sizes = [n // clusters] * clusters
+    prob_matrix = [[intra_p if i == j else inter_p for j in range(clusters)] for i in range(clusters)]
+    return ig.Graph.SBM(sum(block_sizes), prob_matrix, block_sizes)
```

Comment:
Could we simplify code the code while also make it more illustrative and use empirical network data here?

You could try the karate club network, the Les Miserables network (already available in the same directory) or perhaps the famous [Jazz musicians network](https://networks.skewed.de/net/jazz_collab). See which one gives a nicer result.

For the random graph, let's use one that has the same vertex count and density as the empirical one. Measure the density and pass it as the $p$ parameter of the $G(n,p)$ model. Alternatively, measure the edge count and pass it as the $m$ parameter of the $G(n,m)$ model.